src/module-elasticsuite-core/Index/Mapping/Field.php

```diff
@@ -200,14 +200,14 @@ public function getMappingProperty($analyzer = self::ANALYZER_UNTOUCHED)
         $fieldName    = $this->getName();
         $propertyName = $fieldName;
         $property     = $this->getMappingPropertyConfig();
-        $isDefaultAnalyzer = $analyzer == $this->getDefaultSearchAnalyzer();
+        $isDefaultAnalyzer = $analyzer === $this->getDefaultSearchAnalyzer();
 
-        if (isset($property['fields']) && !$isDefaultAnalyzer) {
+        if (!$isDefaultAnalyzer && isset($property['fields'])) {
```

Comment:
why inverting the initial condition ?

Reply:
Condition "!$isDefaultAnalyzer" execution costs less condition "isset($property['fields'])" .

Reply:
That's what I had in mind, good spot. Ok for this one